mlflow/deployments/base.py

```diff
@@ -174,15 +174,15 @@ def get_deployment(self, name):
 
     @abc.abstractmethod
     @experimental
-    def predict(self, deployment_name, df):
+    def predict(self, deployment_name, input, output=None):
```

Comment:
BTW, I'd recommend having the `input` and `output` parameters be CLI only - thanks for adding the CLI by the way! The way you can implement this in the CLI is:
* Load the input JSON file into a pandas dataframe using `df = pd.read_json(input_path)` (see [pandas docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_json.html#pandas-read-json))
* Call the existing `output_df = client.predict(deployment_name, df)` API
* Save the result to the output JSON file via `output_df.to_json(output_path)` (see [pandas docs](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_json.html#pandas-dataframe-to-json))

Reply:
We are working on this change. Will post a PR soon.

Reply:
As suggested, we have commited the code with
1. cli - takes json path as input and writes into the output path
2. base - reverted the code. takes input as dataframe as before
3. serve - input can either be dataframe or json string. 

Please review and let us know your comments.